src/pages/Login/Login.jsx

```diff
@@ -1,5 +1,124 @@
-import React from "react";
+import React, { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth, usePlaylist } from "../../contexts";
+import { loginService } from "../../services";
+import { useAuthForm, useScrollToTop, useDocumentTitle } from "../../hooks";
+import { Input } from "../../components";
 
 export const Login = () => {
-  return <div>Login</div>;
+  const [error, setError] = useState("");
+  const {
+    loading,
+    showPassword,
+    credentials,
+    handleShowPassword,
+    handleInputChange,
+    authFormDispatch,
+  } = useAuthForm({
+    email: "",
+    password: "",
+  });
+
+  const navigate = useNavigate();
+  const { updateUser } = useAuth();
+  const { playlistDispatch } = usePlaylist();
+
+  useScrollToTop();
+  useDocumentTitle("Login");

```

Comment:
These are side effects. Move them to useEffect

Reply:
those are custom hooks, which are using useEffect only